### PR TITLE
[Desktop-lite]- libasound2 not installing in noble - issue

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Feature name**:
 
 * Desktop-lite
 
 **Description**:
 
 This PR introduces the following functionality :

 * As a resolution to Features issue [#968](https://github.com/devcontainers/features/issues/968)
 * Upon introduction of `Ubuntu-24.04 (noble numbat)`, build has been failing because of `libasound2` not being currently available for `Ubuntu-24.04`, so as a result, switching to `libasound2-dev` as an alternative. 
 
 _Changelog_:
 
 * Updated `install.sh`
 * Updated tests to validate the above scenario
 
 **Checklist**:
 
 * [x]   Checked that applied changes work as expected

### CODE REVIEW

File: src/desktop-lite/devcontainer-feature.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "desktop-lite",
-    "version": "1.0.8",
+    "version": "1.0.9",
```
Comment (samruddhikhandale) [inline src/desktop-lite/devcontainer-feature.json:3]:
```suggestion
    "version": "1.1.0",
```

File: src/desktop-lite/install.sh [hunk 2]
```diff
@@ -198,6 +197,15 @@ fi
 # Install X11, fluxbox and VS Code dependencies
 check_packages ${package_list}
 
+# if Ubuntu-24.04 (noble numbat) found, then install libasound2-dev else libasound2
```
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:200]: Can we add a note that this is temporary and we will swtich to `libasound2` once noble is available.

Also, add https://packages.ubuntu.com/noble/libasound2  link in the comment

File: src/desktop-lite/install.sh [hunk 3]
```diff
@@ -198,6 +197,15 @@ fi
...
+. /etc/os-release
+if [ "${ID}" = "ubuntu" ] && [ "${VERSION_CODENAME}" = "noble" ]; then
+    echo "Ubuntu 24.04 Noble (Numbat) detected."
```
Comment (samruddhikhandale) [inline src/desktop-lite/install.sh:203]:
```suggestion
    echo "Ubuntu 24.04 Noble (Numbat) detected. Installing libasound2-dev package..."
```

File: test/desktop-lite/scenarios.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+    "desktop-lite_noble_numbat": {
+        "image": "ubuntu:noble",
```
Comment (samruddhikhandale) [inline test/desktop-lite/scenarios.json:3]: Thanks for adding these tests. However, we already have workflows which are automatically adding these tests. 

Can we remove the scenarios? Thanks!

File: test/desktop-lite/desktop-lite_noble_numbat.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        echo "❌ Package '$PACKAGE_NAME' is not installed."
+    fi
+}
+
+check "desktop-init-exists" bash -c "ls /usr/local/share/desktop-init.sh"
+check "log-exists" bash -c "ls /tmp/container-init.log"
+check "fluxbox-exists" bash -c "ls -la ~/.fluxbox"
+
+checkOSPackage "if libasound2 exists !" "libasound2"
+checkOSPackage "if libasound2-dev exists !" "libasound2-dev"
```
Comment (samruddhikhandale) [inline test/desktop-lite/desktop-lite_noble_numbat.sh:30]: I understand you want to add checks for this, can we update https://github.com/devcontainers/features/blob/5b909b435dcfa46949b95d03858eec747e11be1c/test/desktop-lite/test.sh instead?

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]

File: src/desktop-lite/devcontainer-feature.json (hunk 1 above)
Comment (gauravsaini04) [reply src/desktop-lite/devcontainer-feature.json]: done

File: src/desktop-lite/install.sh (hunk 2 above)
Comment (gauravsaini04) [reply src/desktop-lite/install.sh]: done

File: src/desktop-lite/install.sh (hunk 3 above)
Comment (gauravsaini04) [reply src/desktop-lite/install.sh]: done

File: test/desktop-lite/desktop-lite_noble_numbat.sh (hunk 5 above)
Comment (gauravsaini04) [reply test/desktop-lite/desktop-lite_noble_numbat.sh]: done

File: test/desktop-lite/scenarios.json (hunk 4 above)
Comment (gauravsaini04) [reply test/desktop-lite/scenarios.json]: done 👍

File: test/desktop-lite/test.sh [hunk 6]
```diff
@@ -5,9 +5,29 @@ set -e
...
+        echo "❌ Package '$PACKAGE_NAME' is not installed."
+    fi
+}
+
 check "desktop-init-exists" bash -c "ls /usr/local/share/desktop-init.sh"
 check "log-exists" bash -c "ls /tmp/container-init.log"
 check "fluxbox-exists" bash -c "ls -la ~/.fluxbox"
 
+checkOSPackage "if libasound2 exists !" "libasound2"
+checkOSPackage "if libasound2-dev exists !" "libasound2-dev"
```
Comment (samruddhikhandale) [inline test/desktop-lite/test.sh:30]: how are the tests passing for noble as well as others distros (eg. jsammy) ? 
Shouldn't the check for `libasound2` fail for noble? I don't see OS related condition to make it happy

Comment (gauravsaini04) [reply test/desktop-lite/test.sh]: thanks, done 👍

**Review by samruddhikhandale — APPROVED** [review]
Thanks!

**Pull request merged**